app/controllers/spree/api/ams/orders_controller.rb

```diff
@@ -4,6 +4,15 @@ module Ams
       class OrdersController < Spree::Api::OrdersController
         include Serializable
         include Requestable
+
+        def mine
+          if current_api_user.persisted?
+            @orders = current_api_user.orders.reverse_chronological
+            respond_with(@orders)
+          else
+            render "spree/api/errors/unauthorized", status: :unauthorized
+          end
+        end
```

Comment:
Oops, sorry looks like they use ransack for pagination:

https://github.com/spree/spree/blob/5f3c342fc6d7f4c86ca2f328c6342784948b94b0/api/app/controllers/spree/api/orders_controller.rb#L49

For CanCan this might work

```
authorize! :index, @orders
```

Otherwise, you might need to do something like 

```
@orders.each do |order|
  authorize! :show, order, order.order_token
end
```

This should take care of rendering the `unauthorized` error for you